cerberus-auth-connector-okta/src/main/java/com/nike/cerberus/auth/connector/okta/OktaAuthConnector.java

```diff
@@ -96,6 +99,36 @@ public AuthResponse triggerChallenge(String stateToken, String deviceId) {
     }
   }
 
+  /** Triggers challenge for SMS or Call factors using Okta Auth SDK. */
+  public AuthResponse triggerPush(String stateToken, String deviceId) {
+
+    CompletableFuture<AuthResponse> authResponseFuture = new CompletableFuture<>();
+    PushStateHandler stateHandler =
+        new PushStateHandler(oktaAuthenticationClient, authResponseFuture);
+
+    try {
+      oktaAuthenticationClient.verifyFactor(deviceId, stateToken, stateHandler);
+
+      AuthResponse authResponse = authResponseFuture.get(45, TimeUnit.SECONDS);
+      while (authResponse.getData().getFactorResult().equals("WAITING")) {
```

Comment:
How long will this while loop last? Meaning how long is a push notification good for until it expires?

Reply:
After reading the docs, it looks like Okta will return `TIMEOUT`, so the while loop then would exit.

https://developer.okta.com/docs/reference/api/factors/#factor-verify-result-object

Reply:
Yeah, one thing I'm still not clear on is how long they wait. The UI will wait 60s prior to timing out which I feel is a good amount of time, not sure how long it takes for the okta factor to expire though. Worth exploring so we can ensure we error appropriately.

Reply:
I left it at 60 seconds for time out. The Okta timeout is over 5 minutes so I just don't think it makes sense to wait that long. I also updated the system to let the client know about the lack of response so the errors inform the user.